### ISSUE TITLE
hv.Text and hv.Labels should support `x_units` and `y_units`

### ISSUE THREAD

**Issue by ahuang11** [holoviz/holoviews#6184]
Seems like `x_units="screen"` is supported from https://github.com/bokeh/bokeh/issues/12215

```python
from bokeh.plotting import figure, show
from bokeh.models import Label

fig = figure(title='Label position test')
fig.sizing_mode = 'stretch_both'

fig.circle([1, 2, 3, 4, 5], [6, 7, 2, 4, 5], size=20, color="navy", alpha=0.5)

lower_left = Label(x=5, y=5, x_units='screen', y_units='screen', text='Lower left corner')
fig.add_layout(lower_left)

show(fig)
```

However, holoviews doesn't support it:
```python
import holoviews as hv

hv.Labels([(0, 0, 'A'), (0.5, 0.5, 'B'), (1, 1, 'C')]).opts(x_units="screen")
```

**Comment by TheoMathurin** [comment]
+1 on this one

In the past I've resorted to using a plot hook to add a Bokeh Label glyph.

However at the time I looked into this, it seemed to me that the Labels element uses Bokeh's Text glyph under the hood, and this does not offer the `x_units`/`y_units` options.

**Comment by Hoxbro** [comment]
I don't know if I'm a fan of the name as they collide with the unit in `hv.Dimension`. 

The functionality is definitely +1.

**Comment by ahuang11** [comment]
Wait I don't understand what you mean by collide with the unit hv.Dimesnion?
```python
import holoviews as hv

hv.Dimension("a", unit="m/s^2")
```

```python
import holoviews as hv

hv.Dimension("a", x_units="m/s^2")
```
```
TypeError: Dimension.__init__() got an unexpected keyword argument 'x_units'
```

**Comment by Hoxbro** [comment]
I meant the name itself, not if the code could run. How should a user know that `x_units` is meant to be the unit definition of the figure, not the unit of the x-axis dimension?

**Comment by TheoMathurin** [comment]
It could be confusing but for backend-specific options I always assumed users were expected to use the keywords as exposed by the backend library.

Anyway whatever name is used, it remains that the underlying Bokeh `Text` model does not support it, right? In this case, for this to work it would be necessary to migrate from `Text` to `Label` or `LabelSet`.

**Comment by ahuang11** [comment]
Yeah, I think `Label` makes sense, especially for `hv.Labels` if it's not already.

**Comment by ahuang11** [comment]
Would you be interested in contributing this refactor?

**Comment by TheoMathurin** [comment]
I don't know what such a change would entail. For starters I don't even know where the mapping between hv elements  and bokeh glyphs is done, but I can have a look.

**Comment by ahuang11** [comment]
My understanding is all backends share the same "[element](https://github.com/holoviz/holoviews/blob/main/holoviews/element/annotation.py#L486)" and each backend has its unique "[plot](https://github.com/holoviz/holoviews/blob/main/holoviews/plotting/bokeh/annotation.py)" implementation, calling the `[plot_methods](https://github.com/holoviz/holoviews/blob/main/holoviews/plotting/bokeh/annotation.py#L183)`. 

My only concern is that updating Text -> Labels will introduce breaking changes.

**Comment by ahuang11** [comment]
I am not sure that matters.

```
    def _init_glyph(self, plot, mapping, properties):
        self._plot_methods = {"single": self._methods[self.invert_axes]}
        return super()._init_glyph(plot, mapping, properties)
```

For example Span overrides this init_glyph method:
https://github.com/holoviz/holoviews/blob/main/holoviews/plotting/bokeh/annotation.py#L243-L248

LabelSet seems to accept source.
https://docs.bokeh.org/en/2.4.3/docs/user_guide/annotations.html#labels
